### PR TITLE
show Tokenization

### PR DESCRIPTION
**Pull request by xraxra**
adds prompt argument -t which prints out the tokenized prompt, useful when checking if prompt fits into token limit etc

### CODE REVIEW

**Comment by bmaltais** [comment]
Sound like a great addition. I was always wondering if it was possible to get the actual list of tokens used by SD. Nice work.

File: ldm/simplet2i.py [hunk 1]
```diff
@@ -637,3 +641,27 @@ def _run_gfpgan(self, image, strength):
...
+    # but for readability it has been replaced with ' '
+    def _log_tokenization(self, text):
+        if not self.log_tokenization:
+            return
+        tokens = self.model.cond_stage_model.tokenizer._tokenize(text)
+        tokenized = ""
+        discarded = ""
+        usedTokens = 0
+        totalTokens = len(tokens)
+        for i in range(0,totalTokens):
```
Comment (tildebyte) [inline ldm/simplet2i.py:656]: Can't this simply be `for token in tokens:`, and then actually **remove** the `cond_stage_model.max_length` check below, given that you'll only get as many tokens from the iterator as actually exist...

**Comment by bmaltais** [comment]
I weaved this in the last commit to see and it is AMAZING! This made me realize just how much get discarded in a long prompt. This will make prompt creation SO MUCH better! Good job @xraxra 

I did some testing and the nice thing is that each sub prompt start with a fresh 77 tokens... so sub prompts is the way to go to use longer prompt.

**Comment by tildebyte** [comment]
> sub prompt

How does this work? Is that what commas do?

**Comment by bakkot** [comment]
@tildebyte I assume that's a reference to [this feature](https://github.com/lstein/stable-diffusion#weighted-prompts).

**Comment by xraxra** [comment]
I'm going to close this pull request and open a new one from a clean branch, I messed up some merges and don't like how messy the changes look

**Pull request closed without merging**